doc/sphinx/practical-tools/coqide.rst

```diff
@@ -5,130 +5,165 @@
-message window, and the location of the error is emphasized by an underline in
-the error foreground color (red by default).
-
-In the figure :ref:`CoqIDE main screen <coqide_mainscreen>`,
+-----------------------------------------
+
+The *File* menu lets you open files, create new buffers, save buffers to a file,
+and print or export them into various formats.
+
+Text editing provides the basic operations such as copy/cut/paste and replace.
+Most editing operations are shown in the *Edit* menu, which shows the equivalent
+keystrokes for operations.  if you need more complex editor commands, you can
+launch your favorite text editor on the current buffer, using the
+*Edit/External Editor* menu. (When you're done editing you must currently must
+reopen the file to see your changes.)  Also note:
+
+- Undo is also available as Ctrl-Z.  Redo is Ctrl-Shift-Z.
+- Select all is Ctrl-A.
+- Commenting and uncommenting the current line or selected text is available in
+  the *Tools* menu.
+
+Files are automatically saved periodically to a recovery file.  For example,
+`foo.v` is saved to `#foo.v#` every 10 seconds by default.  You can change the
+interval in the *Edit / Preferences / Files* dialog.
+
+In the *View* menu, you can set almost a dozen printing options that
+correspond to options that can appear in the script.  For example, "Display
+notations" on the menu corresponds to the :flag:`Printing Notations` flag.  You
+should use the menu instead of controlling these settings in your script.
+
+Running Coq scripts
+-------------------
+
+Operations for running the script are available in the *Navigation* menu,
+from the toolbar and from the keyboard.  These include:
+
+- Forward (Ctrl-Down) to run one command or tactic
+- Backward (Ctrl-Up) undo one command or tactic
+- Run to cursor (Ctrl-Right) to run commands up to the cursor
+- Run to end (Ctrl-End) to run commands to the end of the buffer
+- Reset Coq (Ctrl-Home) to restart the Coq process
+- Interrupt to stop processing commands
+
+Commands may:
+
+- Complete successfully.  In this case, the background of the command is marked
+  with the "processed" color (green by default).
+- Complete with a warning.  In this case, the warning appears in the messages
+  panel in yellow.  The background of the command is marked with the "processed"
+  color and the text is shown in blue and underlined.  The message text is
+  available as a tooltip on the text of the command.
+- Fail with an error.  If you're stepping through the proof line by line, the
+  error message appears in the message panel in red and the command is shown
+  in red and underlined with a pink background.  If you're in async mode,
+  described in more detail later, the message appears in the *errors panel*.
+  Double click on an entry to jump to the point of the error.  Execution
+  of commands stops unless you're in async mode.
+
+.. todo: describe the ribbon (a bit broken, though)
```

Comment:
How broken? I think the ribbon is a very useful indicator on the progress of a proof and the fact that you can click it to navigate the buffer is quite useful.

Reply:
I don't know how to explain how to interpret it and its dynamic behavior.  It's animated but doesn't communicate very well.  For example, looking at Pff.v, first I see

![image](https://user-images.githubusercontent.com/1253341/170413436-dcfb689a-b1b3-4b11-b4eb-bcb16f744b67.png)

then a bit later I see

![image](https://user-images.githubusercontent.com/1253341/170413500-02f0fbb5-ac6c-4908-bd54-c60ca21daf36.png)

What do the gray lines mean, eg in Datatypes.v:

![image](https://user-images.githubusercontent.com/1253341/170414203-34cec18b-b664-46c1-b81c-7ffd73d33ebf.png)

It seems to me it's a somewhat unfinished feature.

Reply:
I would think that `Pff.v` is quite an unusual file due to being over 1000 lines, and I would agree that the ribbon feature in this case is not very helpful. It can be confusing that the ribbon only matches what Coq has actually read, and that the rest of the file is still being read. I don' know if there is an easy way to communicate that.

The grey dots are the unchecked Qeds, and should go away after pressing the check button. I believe this is also not the correct colour. This is probably because the colours in the ribbon are hard coded rather than being derived from the style.

For a 300 line file, for instance. It can be a useful indicator, especially if you have any `Axiom`s or `Admitted` proofs since they show up in yellow.

Reply:
@jfehrle Would you like to write a little bit about the ribbon however? I think mentioning that it is clickable might be useful.